src/Filesystem.cc

```diff
@@ -36,466 +30,293 @@
 #include <ignition/common/Util.hh>
 #include <ignition/common/Uuid.hh>
 #include <ignition/common/Console.hh>
-
-#ifndef _WIN32
-#include <dirent.h>
-#include <limits.h>
-#include <climits>
-#else
-#include <shlwapi.h>
-#include <io.h>
-#include "win_dirent.h"
-#include "PrintWindowsSystemWarning.hh"
-#endif
-
 #include "ignition/common/Filesystem.hh"
 
-#ifdef _WIN32
-# define IGN_PATH_MAX _MAX_PATH
-#elif defined(PATH_MAX)
-# define IGN_PATH_MAX PATH_MAX
-#elif defined(_XOPEN_PATH_MAX)
-# define IGN_PATH_MAX _XOPEN_PATH_MAX
+namespace fs = std::filesystem;
+
+#ifndef _WIN32
+static const char preferred_separator = fs::path::preferred_separator;
 #else
-# define IGN_PATH_MAX _POSIX_PATH_MAX
+static const char preferred_separator = '\\';
```

Comment:
Do we need these static separators? What's the difference between this and the `::separator` function?

Also

```suggestion
static const char preferredSeparator = '\\';
```

Reply:
Not sure, I think it was something windows related, I'll try to drop it and see what happens.